src/Sentry/ScopeExtensions.cs

```diff
@@ -146,109 +145,32 @@ public static bool HasUser(this IScope scope)
-                    (scope.ScopeOptions?.MaxBreadcrumbs ?? Constants.DefaultMaxBreadcrumbs) + 1;
-
-                if (overflow > 0)
-                {
-                    breadcrumbsQueue.TryDequeue(out _);
-                }
-
-                breadcrumbsQueue.Enqueue(breadcrumb);
-            }
-        }
-
-        /// <summary>
-        /// Sets the fingerprint to the <see cref="IScope"/>.
+        /// Sets the fingerprint to the <see cref="Scope"/>.
         /// </summary>
         /// <param name="scope">The scope.</param>
         /// <param name="fingerprint">The fingerprint.</param>
-        public static void SetFingerprint(this IScope scope, IEnumerable<string> fingerprint)
-            => scope.Fingerprint = fingerprint;
-
-        /// <summary>
-        /// Sets the extra key-value to the <see cref="IScope"/>.
-        /// </summary>
-        /// <param name="scope">The scope.</param>
-        /// <param name="key">The key.</param>
-        /// <param name="value">The value.</param>
-        public static void SetExtra(this IScope scope, string key, object? value)
-        {
-            if (scope.Extra is IDictionary<string, object?> extra)
-            {
-                extra[key] = value;
-            }
-        }
+        public static void SetFingerprint(this IEventLike scope, IEnumerable<string> fingerprint)
+            => scope.Fingerprint = fingerprint as IReadOnlyList<string> ?? fingerprint.ToArray();
 
         /// <summary>
-        /// Sets the extra key-value pairs to the <see cref="IScope"/>.
+        /// Sets the extra key-value pairs to the <see cref="Scope"/>.
         /// </summary>
         /// <param name="scope">The scope.</param>
         /// <param name="values">The values.</param>
-        public static void SetExtras(this IScope scope, IEnumerable<KeyValuePair<string, object?>> values)
+        public static void SetExtras(this IEventLike scope, IEnumerable<KeyValuePair<string, object?>> values)
         {
             foreach (var (key, value) in values)
             {
                 scope.SetExtra(key, value);
             }
         }
 
-        /// <summary>
-        /// Sets the tag to the <see cref="IScope"/>.
-        /// </summary>
-        /// <param name="scope">The scope.</param>
-        /// <param name="key">The key.</param>
-        /// <param name="value">The value.</param>
-        public static void SetTag(this IScope scope, string key, string value)
-        {
-            if (scope.Tags is IDictionary<string, string> tags)
-            {
-                tags[key] = value;
-            }
-        }
-
         /// <summary>
         /// Set all items as tags.
         /// </summary>
         /// <param name="scope">The scope.</param>
         /// <param name="tags"></param>
-        public static void SetTags(this IScope scope, IEnumerable<KeyValuePair<string, string>> tags)
+        public static void SetTags(this IEventLike scope, IEnumerable<KeyValuePair<string, string>> tags)
```

Comment:
We should rename all variables too now then? Isn't this a refactor on the IDE?

same in other methods of this change
```suggestion
        public static void SetTags(this IEventLike eventLike, IEnumerable<KeyValuePair<string, string>> tags)
```

Reply:
Are you okay with `IEventLike` as the name? It was something I put without thinking, so maybe you have better ideas.